File: src/main/scala/part2test/SelectionManager.scala
package part2test

class SelectionManager {
  private var selectionActive = false
  private var selectedTile:Tile = null

  def selectTile(tile: Tile, runnable: () => Unit): Unit = {
    if (selectionActive) {
      selectionActive = false
//      swap(selectedTile, tile)
      runnable.apply()
    }
    else {
      selectionActive = true
      selectedTile = tile
    }
  }

  private def swap(t1: Tile, t2: Tile): Unit = {
    val pos = t1.getCurrentPosition
    t1.setCurrentPosition(t2.getCurrentPosition)
    t2.setCurrentPosition(pos)
  }
}

File: src/main/java/part2/rmi/SelectionManager.java
package part2.rmi;

public class SelectionManager {

	private boolean selectionActive = false;
	private Tile selectedTile;
	private Tile selectedTile2;

	public void selectTile(final Tile tile, final Runnable runnable) {
		
		if(selectionActive) {
			if (tile.equals(selectedTile)) {
				selectedTile = null;
				selectionActive = false;
			} else {
				selectionActive = false;
				selectedTile2 = tile;
				runnable.run();
			}
		} else {
			selectionActive = true;
			selectedTile = tile;
		}
	}

	public void swap() {
		int pos = selectedTile.getCurrentPosition();
		selectedTile.setCurrentPosition(selectedTile2.getCurrentPosition());
		selectedTile2.setCurrentPosition(pos);
	}

}
